frame/vamm/src/tests/update_twap.rs

```diff
@@ -223,26 +262,62 @@ proptest! {
 	#[test]
 	fn update_twap_proptest_succeeds(
 		vamm_state in any_vamm_state(),
-		(base_twap, quote_twap) in any_new_twap()
+		twap in any_new_twap()
 	) {
 		let now = vamm_state.twap_timestamp
-							.saturating_add(1)
-							.min(Timestamp::MAX/1000);
+							.min(Timestamp::MAX/1000)
+							.saturating_add(1);
 		let vamm_state = VammState {
 			closed: None,
 			twap_timestamp: vamm_state.twap_timestamp
-										.min(now).saturating_sub(1),
+										.min(now)
+										.saturating_add(1),
 			twap_period: vamm_state.twap_timestamp
 										.saturating_add(1),
 			..vamm_state
 		};
+
+		let delay = vamm_state.twap_timestamp
+							.saturating_add(vamm_state.twap_period)
+							.saturating_add(1);
+
+		let twap_reciprocal = TestPallet::reciprocal_twap(&twap);
+		assert_ok!(twap_reciprocal);
+
 		ExtBuilder { vamm_count: 1, vamms: vec![(0, vamm_state)] }
 			.build()
 			.execute_with(|| {
-				run_for_seconds(now);
+				run_for_seconds(delay);
 				assert_ok!(
-					TestPallet::update_twap(0, base_twap, quote_twap),
+					TestPallet::update_twap(0, Some(twap), twap_reciprocal.ok()),
+					(twap, twap_reciprocal.unwrap())
+				);
+
+				run_for_seconds(delay);
+				assert_ok!(
+					TestPallet::update_twap(0, Some(twap), None)
+				);
+
+				run_for_seconds(delay);
+				assert_ok!(
+					TestPallet::update_twap(0, None, Some(twap))
+				);
+
+				run_for_seconds(delay);
```

Comment:
I think these may suffer from the same issue pointed out in the other comment

Reply:
yeah, I changed it too. 
Thanks for pointing out! :)